Prep version 3.2.0 (#6187)
sqlfluff/sqlfluff: CHANGELOG.md

-->
<!--Start Of Releases (DO NOT DELETE THIS LINE)-->

## [3.2.0] - 2024-09-15

## Highlights

This release brings a few minor breaking changes, both for the core project
and for the dbt templater. For the main project:

> * Resolving an issue with the spacing of functions (LT01), which however

Reviewer: ```suggestion
* Resolving an issue with the spacing of functions (LT01), which
```

  involved a change to how functions are parsed. If your project relies
  on the specific parsing of functions, the bracketed arguments are now
  wrapped in a `function_contents` object. We recommend that you examine
  the new parsing structure using this new release in testing first.

* `RF06` (`references.quoting`) is now case sensitive when removing
  quotes which are detected as unnecessary. This rule has also been re-enabled
  by default for Snowflake and Postgres where it had previously been disabled
  (for the reason that in the past it hadn't been appropriately case
  sensitive). Treatment for totally case-insensitive dialects like DuckDB
  and SparkSQL have also been included. Please check the new documentation
  for this rule (which is much more explicit now), for details related to
  your dialect.

* Significant changes under the hood to the handling of configuration files.

>   Most of these should now be visible to end users, but for anyone integrating

Reviewer 2: ```suggestion
  Most of these should be invisible to end users, but for anyone integrating
```

>   SQFluff into a larger project and relying on native file loading may need

Reviewer 2: ```suggestion
  SQLFluff into a larger project and relying on native file loading may need
```
